src/core/net/dnssd_server.cpp

```diff
@@ -867,6 +867,11 @@ void Server::OnUpstreamQueryDone(UpstreamQueryTransaction &aQueryTransaction, Me
         error = mSocket.SendTo(*aResponseMessage, aQueryTransaction.GetMessageInfo());
     }
 
+    if (error == kErrorNone)
+    {
+        mCounters.mHandledByUpstream.mForwardedResponses++;
+    }
+
     ResetUpstreamQueryTransaction(aQueryTransaction, error);
```

Comment:
Maybe even better to update the counters in ` ResetUpstreamQueryTransaction()` which gets the `error`?